pynetbox/core/response.py

```diff
@@ -265,6 +268,23 @@ def list_parser(list_item):
                 self._add_cache((k, v))
             setattr(self, k, v)
 
+    def _parse_related_item(self, values):
+        """ Parses values for a related item contained within
+        the init values.
+
+        Returns:
+            Record object of the appropriate type
+            with the correct Endpoint,
+        """
+        url = values["url"]
+        app, name, _ = url.strip("/").split("/")[-3:]
+        app = getattr(self.api, app)
+        model = app.model
+
+        endpoint = self.endpoint.__class__(self.api, app, name, model)
+
+        return endpoint.return_obj(values, self.api, endpoint)
```

Comment:
Leverage the existing logic implemented in the endpoint to resolve model classes etc.